orchestration/dagster_orchestration/hca_orchestration/contrib/argo_workflows.py

```diff
@@ -0,0 +1,67 @@
+from argo.workflows.client import ApiClient as ArgoApiClient,\
+                                  ArchivedWorkflowServiceApi,\
+                                  Configuration as ArgoConfiguration
+from argo.workflows.client.models import V1alpha1Workflow
+
+
+def generate_argo_archived_workflows_client(host_url, access_token):
+    return ArchivedWorkflowServiceApi(
+        api_client=ArgoApiClient(
+            configuration=ArgoConfiguration(host=host_url),
+            header_name="Authorization",
+            header_value=f"Bearer {access_token}"))
+
+
+class ArgoArchivedWorkflowsClientMixin:
+    def __init__(self, *args, argo_url, access_token, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.argo_url = argo_url
+        self.access_token = access_token
+        self._client = None
+
+    def client(self):
+        if not self._client:
+            self._client = generate_argo_archived_workflows_client(self.argo_url, self.access_token)
+
+        return self._client
+
+    def list_archived_workflows(self):
+        self._pull_paginated_results(self.client().list_archived_workflows)
+
+    def _pull_paginated_results(self, api_function):
+        results = api_function()
+
+        for result in results.items:
+            yield result
+
+        while results.metadata['_continue']:
+            results = api_function(list_option_continue=results.metadata['_continue'])
```

Comment:
What happens with the sensor when argo inevitably 500/502s and this raises an exception?

Reply:
the run dies and produces no run requests, but it also won't prevent a future run from picking up the run requests that should have been generated by the prior failed run.

Reply:
probably worth adding error handling here, but unsurprisingly the docs for the autogenerated python client for argo's api are pretty sparse about errors that can be raised by various actions